### PR TITLE
Wasm test fixes

### PR DESCRIPTION
**Pull request by gneeri**
> [!NOTE]
> **Medium Risk**
> Touches execution semantics in the REPL/VM (auto-output capture and relational dispatch), MATLAB parsing of matrix literals, and filesystem path resolution on `wasm32`, which could change observable behavior in edge cases and tests.
> 
> **Overview**
> Improves MATLAB-compatibility and wasm test reliability by adjusting how final results are surfaced when output isn’t suppressed (including new fallbacks to recover the last emitted value/assignment), and by routing relational operators through builtins for non-numeric types.
> 
> Fixes parsing of matrix literals to better match MATLAB whitespace rules (notably `[1 +2]` vs `[1 + 2]` and imaginary literals), and updates `wasm32` filesystem behavior to resolve relative paths via a managed `current_dir` (with `current_dir`/`set_current_dir` APIs) so file operations behave consistently.
> 
> Also tightens/extends a few runtime semantics and display behaviors: `cat` can build char arrays from scalar code points, `flip([])` defaults instead of treating empty dims as explicit, cell/string/logical displays are more MATLAB-like (including N-D shape summaries), GC root scanning now recurses through nested cells/structs, and docs/benchmarks are tweaked for runnable examples and expected outputs.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 28fe59069c7c44ff05553795388053ce9fec41a4. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: crates/runmat-filesystem/src/lib.rs [hunk 1]
```diff
@@ -261,18 +262,42 @@ impl Seek for File {
...
+#[cfg(target_arch = "wasm32")]
+static CURRENT_DIR: OnceCell<RwLock<PathBuf>> = OnceCell::new();
 
 fn provider_lock() -> &'static RwLock<Arc<dyn FsProvider>> {
     PROVIDER.get_or_init(|| RwLock::new(default_provider()))
 }
 
+#[cfg(target_arch = "wasm32")]
+fn current_dir_lock() -> &'static RwLock<PathBuf> {
+    CURRENT_DIR.get_or_init(|| RwLock::new(PathBuf::from("/Users/testUser/runmat-project")))
```
Comment (nallana) [inline crates/runmat-filesystem/src/lib.rs]: This seems odd

File: crates/runmat-gc/src/roots.rs [hunk 2]
```diff
@@ -100,14 +100,22 @@ impl GcRoot for StackRoot {
...
+                    let inner = unsafe { &*cell_value.as_raw() };
+                    self.collect_value_roots(inner, roots);
+                }
             }
+            Value::Struct(struct_value) => {
+                for field_value in struct_value.fields.values() {
+                    self.collect_value_roots(field_value, roots);
+                }
+            }
+            _ => {}
```
[the remark this answers is not in the document]
Comment (nallana) [reply crates/runmat-gc/src/roots.rs]: This is a good catch. This should be centralized

[the remark this answers is not in the document]
Comment (nallana) [reply crates/runmat-gc/src/roots.rs]: Another good catch

File: crates/runmat-parser/src/lib.rs [hunk 3]
```diff
@@ -1058,6 +1060,31 @@ impl Parser {
...
+                            matches!(info.token, Token::Ident)
+                                && (info.lexeme.eq_ignore_ascii_case("i")
+                                    || info.lexeme.eq_ignore_ascii_case("j"))
+                                && self.tokens_adjacent(rhs_index, rhs_index + 1)
+                        })
+                        .unwrap_or(false);
+                if !rhs_is_imag_literal {
+                    break;
+                }
+            }
```
[the remark this answers is not in the document]
Comment (nallana) [reply crates/runmat-parser/src/lib.rs:1092]: Another good catch

File: crates/runmat-runtime/src/builtins/array/shape/cat.rs [hunk 4]
```diff
@@ -422,18 +422,38 @@ fn cat_char_arrays(dim_zero: usize, values: Vec<Value>) -> BuiltinResult<Value>
...
+    if !value.is_finite() || value.fract() != 0.0 {
+        return Err(cat_err("cat: expected integer code point for char concatenation"));
+    }
+    if value < 0.0 || value > u32::MAX as f64 {
+        return Err(cat_err("cat: code point out of range"));
+    }
+    let code = value as u32;
+    let ch = char::from_u32(code).ok_or_else(|| cat_err("cat: invalid code point"))?;
+    CharArray::new(vec![ch], 1, 1).map_err(|e| cat_err(format!("cat: {e}")))
+}
```
[the remark this answers is not in the document]
Comment (nallana) [reply crates/runmat-runtime/src/builtins/array/shape/cat.rs]: Should also be implemented

File: crates/runmat-runtime/src/builtins/common/env.rs [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+#[cfg(target_arch = "wasm32")]
+use std::collections::BTreeMap;
+#[cfg(target_arch = "wasm32")]
+use std::sync::RwLock;
+
+#[cfg(target_arch = "wasm32")]
+use once_cell::sync::OnceCell;
+
+#[cfg(target_arch = "wasm32")]
+fn env_lock() -> &'static RwLock<BTreeMap<String, String>> {
```
Comment (nallana) [inline crates/runmat-runtime/src/builtins/common/env.rs:12]: Would be good to chat through some of this part on Monday
